pkg/checkpoint/runtime_service.go

```diff
@@ -93,53 +88,123 @@ func (r *remoteRuntimeService) localRunningPods() map[string]*v1.Pod {
 	return pods
 }
 
-func (r *remoteRuntimeService) getRunningKubeletContainers() ([]*runtimeapi.Container, error) {
-	filter := &runtimeapi.ContainerFilter{}
-
-	// Filter out non-running containers
-	filter.State = &runtimeapi.ContainerStateValue{
-		State: runtimeapi.ContainerState_CONTAINER_RUNNING,
-	}
-
-	return r.listContainers(filter)
+type criContainer struct {
+	Labels map[string]string
 }
 
-func (r *remoteRuntimeService) getRunningKubeletSandboxes() ([]*runtimeapi.PodSandbox, error) {
-	filter := &runtimeapi.PodSandboxFilter{}
-
-	// Filter out non-running sandboxes
-	filter.State = &runtimeapi.PodSandboxStateValue{
-		State: runtimeapi.PodSandboxState_SANDBOX_READY,
-	}
-	return r.listPodSandbox(filter)
+type criSandbox struct {
+	Uid       string
+	Name      string
+	Namespace string
+	Labels    map[string]string
 }
 
-func (r *remoteRuntimeService) listPodSandbox(filter *runtimeapi.PodSandboxFilter) ([]*runtimeapi.PodSandbox, error) {
+func (r *remoteRuntimeService) getRunningKubeletContainers() ([]criContainer, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), r.timeout)
 	defer cancel()
 
-	resp, err := r.runtimeClient.ListPodSandbox(ctx, &runtimeapi.ListPodSandboxRequest{
-		Filter: filter,
+	var containers []criContainer
+	if _, err := r.v1alpha1Client.Version(ctx, &v1alpha1.VersionRequest{}); err == nil {
+		resp, err := r.v1alpha1Client.ListContainers(ctx, &v1alpha1.ListContainersRequest{
+			Filter: &v1alpha1.ContainerFilter{
+				State: &v1alpha1.ContainerStateValue{
+					// Filter out non-running containers
+					State: v1alpha1.ContainerState_CONTAINER_RUNNING,
+				},
+			},
+		})
+		if err != nil {
+			glog.Errorf("ListContainers with filter from runtime service failed: %v", err)
+			return nil, err
+		}
+		for _, c := range resp.Containers {
+			if c.Metadata == nil {
+				glog.V(4).Infof("Container does not have metadata: %+v", c)
```

Comment:
Why do we skip here? When does this happen?

Reply:
Just copied from the previous behavior.

Reply:
Yeah, I saw the removed block too, was just curious. not a big deal.